### DISCUSSION TITLE
엔티티 필드 중 열거체 (e.g. 주문 상태, 결제 수단, 펀딩 상태 등) 코드 통일

### DISCUSSION THREAD

**Discussion by kmw2378** [KakaoFunding/back-end#61] [Ideas]
이번 결제 기능을 구현하면서 `String`으로 정의되있던 주문 상태, 결제 수단 필드를 모두 열거체화 했습니다. 이렇게 열거체화가 되있지 않은 필드가 아직 남아있는데 맡은 분마다 다르게 열거체를 작성하면 헷갈릴거 같아 아래와 같은 규칙을 정하는건 어떨까요??

__OrderType.java__
```java
@Getter
public enum OrderStatus {
    COMPLETE_PAYMENT("결제 완료"),
    PURCHASE_DECISION("구매 결정"),
    CANCELLATION_RETURN_EXCHANGE("취소/환불/교환");

    private final String description;

    OrderStatus(final String description) {
        this.description = description;
    }
}
```
- 필드명은 `description`으로 통일
- 그 외에 필드가 필요하다면 팀원과 상의 후 결정

**Comment by YeaChan05** [comment]
확실히 문자열이 그냥 노출되는 경우가 많아서 열거체로 처리해야한다는 생각은 있었는데 매번 귀찮아서 미루긴 한거같네요 ㅎㅎ...
URL path같은것도 열거체 처리하는건 어떠실까요?

**Reply by kmw2378** [reply]
URL path에 적용하는건 신선하긴 하네요...! `SecurityConfig`, Controller 등 다양하게 사용되니 이를 열거체 하나로 관리하는 것도 정말 좋은 방법인 거 같습니다!

**Comment by hwan2-99** [comment]
저도 펀딩상태를 혼자 enum으로 만들기엔 좀 별로일것같아서 일단 문자열로 처리하긴했는데 좋은 생각인것 같습니다!!
